pkg/etcdmanager/etcdmanager.go

```diff
@@ -0,0 +1,560 @@
+/*
+/ SPDX-FileCopyrightText: 2021 Finanz Informatik Technologie Services GmbHs
+/
+/ SPDX-License-Identifier: AGPL-1.0-only
+*/
+
+package etcdmanager
+
+import (
+	"context"
+	errs "errors"
+	"fmt"
+	"os"
+	"strings"
+
+	pg "github.com/fi-ts/postgreslet/api/v1"
+	"github.com/go-logr/logr"
+	coreosv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	// EtcdComponentLabelName Name of the managed-by label
+	EtcdComponentLabelName string = "postgres.database.fits.cloud/component"
+	// EtcdComponentLabelValue Value of the managed-by label
+	EtcdComponentLabelValue string = "etcd"
+)
+
+// Options
+type Options struct {
+	EtcdImage              string
+	EtcdBackupSidecarImage string
+	PostgresletNamespace   string
+	PartitionID            string
+	SecretKeyRefName       string
+	PSPName                string
+	PostgresletFullname    string
+}
+
+// OperatorManager manages the operator
+type EtcdManager struct {
+	client.Client
+	runtime.Decoder
+	list *corev1.List
```

Comment:
this is actually the deserialized etcd manfest, therefore the field name is wrong